### ISSUE TITLE
List the main person(s) working in each ETL step

### ISSUE THREAD

**Issue by pabloarosado** [owid/etl#2465]
## Problem

We currently have no easy way to know who has worked in a particular ETL step or grapher dataset. 

If, for example, something odd is found in the data, we don't know straight away who the contact person is, to tag in an issue or in slack.

Hence, it would be convenient to have that information both in the ETL dashboard and in the grapher admin.

### Add a metadata field

One possible solution would be to have `owid_authors` (or some other name) as a new field (list of literals) in the snapshot metadata, which would be automatically propagated (by doing the union of authors). As with any other metadata field, this field could be manually overwritten at a later step.

### Use git metadata

We could check who's committed to a step. It would have the benefit of filling this in historically, but we also have bots that commit, and many ingredients on the way to a step. But it also might pick up people who are not the main contributor, e.g. Mojmir gets tagged on EVERYTHING 🎉 

## What's in scope

- [ ] Create a metadata field
- [ ] Ensure it gets populated automatically (e.g. in Wizard, or from previous data version)
- [ ] Ensure that it propagates, e.g. from snapshot onwards
- [ ] Telling people about it

## Out of scope

- Backfilling it somehow is just nice to have

**Comment by Marigold** [comment]
Reopening as I have a PR https://github.com/owid/etl/pull/3786 ready to be reviewed.

**Comment by Marigold** [comment]
There's an [open PR](https://github.com/owid/etl/pull/3786) for this issue. It'd take a day to get this over the finish line. The question is - do we even need it? No one has missed this feature since we opened the issue. What do you think @pabloarosado?

**Comment by pabloarosado** [comment]
Thanks @Marigold I just commented on [that PR](https://github.com/owid/etl/pull/3786). I don't think it's the most important feature, but if it's an optional metadata field, it can be useful for some future use cases.

**Comment by Marigold** [comment]
> it can be useful for some future use cases.

In that case, we can park it for a bit longer until we find a valuable use case.

**Comment by ikesau** [comment]
Bumping this for triage again - we'd like to get this data bubbled up to our data pages rework this cycle so that we can celebrate and showcase the work that our data scientists do.

Here's a provisional mockup to give you a sense of what we're thinking about:

<img width="978" height="357" alt="Image" src="https://github.com/user-attachments/assets/4e659ca4-a1e7-4b52-8ed6-a80253dd3247" />


I'm assuming that most indicators have 1-2 people behind them - is that reasonable? 

As for a heuristic to backfill these fields for all our indicators, Mojmir (integral as he is) shouldn't be on every single page because of a linter he ran 2 years ago 😉

Feel free to shatter my hopes and dreams here if I'm misconceiving the feasibility of this. Also feel free to ask any questions and I'll do my best to get answers (or join the `#proj-data-pages-r-and-d` channel)

**Comment by lucasrodes** [comment]
@ikesau thanks for reviving this issue! That'd be a cool addition to data pages.

I think we could find ways to do this semi-manually, maybe start with big, important datasets. 

One thing I wonder is what we'd do in cases where `person A` creates a dataset and then leaves OWID. Although _created_ by that person, that dataset is probably no longer managed by that person.

---

For data folks, let's discuss this next week! My sense is that we need a field at the Dataset level. Ideally, at the indicator level, but that's probably not feasible; we can assume all indicators within a dataset are managed by the same people.

We could add the available options (names) into the dataset schema so that we make sure to have harmonized names. I don't know how that flows downstream until a data page atm.

**Comment by Marigold** [comment]
We had a simple PoC in [this PR](https://github.com/owid/etl/pull/3786) that never got merged, but it is quite simple and elegant in my opinion (at least as a starting point).

**Comment by ikesau** [comment]
Yeah, I think there's an emphasis on "maintained by" versus "created by" so that we have a person who's accountable in case errors are found.

The maximally complex solution would be to have both - a list of everyone who's worked on the data, and a single name for the person to talk to if something's wrong with it. 

I'm thinking about Veronika's datasets specifically - it would be nice to have the option to acknowledge that she was the person who made them (even if someone else is their custodian now)

Maybe it could be "whoever's _last_ in the list is the point person" ?

**Comment by pabloarosado** [comment]
Hi @ikesau thanks for bringing this back to life!
Mojmir's proposal would already be a working solution. But there are still some things we'd need to discuss a bit (e.g. previous vs current ownership). We'll talk about it today during our weekly data architecture chat (at 15h CET). Let me know if you'd like to join (no worries otherwise).

**Comment by pabloarosado** [comment]
After our call we decided that:
* Mojmir's [PR](https://github.com/owid/etl/pull/3786) is a good solution, and it makes sense to have a list of "data owners" (the name could be different). The first person there should be the main owner, accountable for the data page, and contact person in case there are issues on that page. So, grapher could signal that somehow.
  * Even though it makes sense to keep the data owner at the indicator level, in practice, in the vast majority of cases, the dataset owner is going to be the same as the owner of all indicators in that dataset. So, to begin with, we can start assigning data owners to datasets.
* Mojmir will take care of the big refactor, and maybe it's possible to create a PR per data owner, so each person can review their list of assigned datasets.
* What happens with historical contributions (e.g. of people who no longer work at OWID)? We have various options:
  * (A) Single list: We keep one list of owners, and it contains anyone who has contributed substantially to that data. We just need to ensure that the first person in the list is actively working at OWID.
  * (B) Two lists: We explicitly make the distinction between owners and authors. It may be an overkill because in most cases they will be identical; but in some cases they'll differ (e.g. for Veronika's datasets).
  * (C) We remove people from those lists once they leave OWID. I think this is not anyone's preferred choice. For articles, we keep authors even after they have left OWID (e.g. Saloni), which I think is good practice. The order of authors may change later on, based on who the main author currently is, but there's no strong need to remove anyone from that list.

**Comment by pabloarosado** [comment]
> _Written by Claude Fable 5 — @pabloarosado at the wheel._

Closing this as completed. The `owners` field was added to the dataset metadata in #6143 (with #3786 as the earlier proof of concept), following the plan agreed above. Existing datasets have been backfilled with owners, the Wizard auto-fills the field for new datasets, and owners propagate to the `datasets` table in the grapher database, so they can be consumed downstream (e.g. by data pages). Thanks everyone!
